terraform/modules/prober/prober.tf

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_cloud_run_v2_job" "jvs_prober" {
+
+  project = var.project_id
+
+  name = "jvs-prober"
+
+  location = "us-central1"
+
+  template {
+
+    template {
+
+      containers {
+        image = var.jvs_prober_image
+
+        env {
+          name  = "JVSCTL_SERVER_ADDRESS"
+          value = var.jvs_service_address
+        }
+
+        env {
+          name  = "JVSCTL_JWKS_ENDPOINT"
+          value = var.jvs_jwks_endpoint
+        }
+
+        env {
+          name  = "AUDIENCE"
+          value = var.jvs_audience
+        }
+      }
+    }
+  }
+
+  lifecycle {
+    ignore_changes = [
+      launch_stage,
+    ]
+  }
+}
+
+# This is the service account which will be used by scheduler_jon
+# to trigger jvs-prober cloud run job.
+data "google_compute_default_service_account" "default" {
+  project = var.project_id
+}
+
+
+# This is the scheduler for triggering jvs-prober cloud run job
+# in a user defined frequency
+resource "google_cloud_scheduler_job" "job" {
+
+  project = var.project_id
+
+  schedule    = var.prober_scheduler
+  name        = "jvs-prober-scheduler"
+  description = "prober cloud run job scheduler"
+  region      = "us-central1"
+
+  retry_config {
+    retry_count = 0
+  }
+
+  http_target {
+    http_method = "POST"
+    uri         = "https://${resource.google_cloud_run_v2_job.jvs_prober.location}-run.googleapis.com/apis/run.googleapis.com/v1/namespaces/${var.project_number}/jobs/${resource.google_cloud_run_v2_job.jvs_prober.name}:run"
```

Comment:
Should `${var.project_number}` be project ID instead? Looking at the example here: https://cloud.google.com/run/docs/execute/jobs-on-schedule#using-scheduler